flink-doris-connector/src/main/java/org/apache/doris/flink/source/reader/DorisFlightValueReader.java

```diff
@@ -0,0 +1,184 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.flink.source.reader;
+
+import org.apache.arrow.adbc.core.AdbcConnection;
+import org.apache.arrow.adbc.core.AdbcDatabase;
+import org.apache.arrow.adbc.core.AdbcDriver;
+import org.apache.arrow.adbc.core.AdbcException;
+import org.apache.arrow.adbc.core.AdbcStatement;
+import org.apache.arrow.adbc.driver.flightsql.FlightSqlDriver;
+import org.apache.arrow.flight.Location;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.doris.flink.cfg.DorisOptions;
+import org.apache.doris.flink.cfg.DorisReadOptions;
+import org.apache.doris.flink.exception.DorisException;
+import org.apache.doris.flink.exception.IllegalArgumentException;
+import org.apache.doris.flink.exception.ShouldNeverHappenException;
+import org.apache.doris.flink.rest.PartitionDefinition;
+import org.apache.doris.flink.rest.RestService;
+import org.apache.doris.flink.rest.SchemaUtils;
+import org.apache.doris.flink.rest.models.Schema;
+import org.apache.doris.flink.serialization.RowBatch;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static org.apache.doris.flink.util.ErrorMessages.SHOULD_NOT_HAPPEN_MESSAGE;
+
+public class DorisFlightValueReader extends ValueReader implements AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(DorisFlightValueReader.class);
+    protected AdbcConnection client;
+    protected Lock clientLock = new ReentrantLock();
+
+    private final PartitionDefinition partition;
+    private final DorisOptions options;
+    private final DorisReadOptions readOptions;
+    private AdbcStatement statement;
+    protected RowBatch rowBatch;
+    protected Schema schema;
+    AdbcStatement.QueryResult queryResult;
+    protected ArrowReader arrowReader;
+    protected AtomicBoolean eos = new AtomicBoolean(false);
+
+    public DorisFlightValueReader(
+            PartitionDefinition partition,
```

Comment:
I think arrow flight is a new reading method, should we no longer rely on the old reading method?

`PartitionDefinition` This class was created using the old reading method. Should we give up using this class?

Reply:
Obtaining the be node arrow readerInfo instance through ticket returns an exception, so the old reading form is temporarily used